OpenAI.SDK/OpenAiOptions.cs

```diff
@@ -22,15 +63,80 @@ public class OpenAiOptions
         /// </summary>
         public string ApiKey { get; set; } = null!;
 
-        public string ApiVersion { get; set; } = "v1";
+        /// <summary>
+        ///     Default Api Version
+        /// </summary>
+        public string ApiVersion
+        {
+            get
+            {
+                return _apiVersion ??= ProviderType switch
+                {
+                    ProviderType.OpenAi => OpenAiDefaultApiVersion,
+                    ProviderType.Azure => AzureOpenAiDefaultApiVersion,
+                    _ => throw new ArgumentOutOfRangeException()
+                };
+            }
+            set => _apiVersion = value;
+        }
+
+        /// <summary>
+        ///     Base Domain
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException"></exception>
+        public string BaseDomain
+        {
+            get
+            {
+                return _baseDomain ??= ProviderType switch
+                {
+                    ProviderType.OpenAi => OpenAiDefaultBaseDomain,
```

Comment:
It seems to me like having the settings behave the same for each type is more important than what you're solving for.  For reference, the way I had it originally is the way the official python library does it.